src/lib/Underground.js

```diff
@@ -168,8 +168,9 @@ class AutomationUnderground
                                  + Automation.Menu.TooltipSeparator
                                  + "Sell-locked items, in the Treasures tab, will never be sold,\n"
                                  + "but might be aquired through tradings.";
-        Automation.Menu.addLabeledAdvancedSettingsToggleButton(
+        const tradeDiamondsButton = Automation.Menu.addLabeledAdvancedSettingsToggleButton(
```

Comment:
I'm not fond of this, it should only trigger when the feature is turned on, otherwise it could get messy.

You should probably consider an approach similar to what was done for the inner loop instead (ie. instanciate the loop interval, if it does not already exist, in the main loop)

Reply:
Oh, you mean when the whole Mine automation is turned on? Yea, that makes sense, I did not think that this would make it run ignoring that

Reply:
Or, you can just call the function manually once right after the setInterval, it will run it when the feature is first enabled and then every 5min

Reply:
That would make it take up to the 5 minutes to run for the first time if the mine is already enabled (and my solution would make it take 5 minutes if the mine was not already enabled)

Reply:
hum, why not